File: Практика_самостоятельная работа/Python/2.21.py
import math

# Функция для вычисления a
def calculate_a(e, f):
    return (e + f / 2) / 3

# Функция для вычисления b
def calculate_b(h, g):
    return abs(h**2 - g)

# Функция для вычисления c
def calculate_c(g, h, e):
    return math.sqrt((g - h)**2 - 3 * math.sin(e))

# Ввод значений e, f, g и h
e = float(input("Введите e: "))
f = float(input("Введите f: "))
g = float(input("Введите g: "))
h = float(input("Введите h: "))

# Вычисление и вывод результатов
a = calculate_a(e, f)
b = calculate_b(h, g)
c = calculate_c(g, h, e)
print("a =", a)
print("b =", b)
print("c =", c)

File: Практика_самостоятельная работа/C/2.21.c
#include <stdio.h>
#include <math.h>

double calculate_a(double e, double f) {
    return (e + f / 2) / 3;
}

double calculate_b(double h, double g) {
    return fabs(h * h - g);
}

double calculate_c(double g, double h, double e) {
    return sqrt((g - h) * (g - h) - 3 * sin(e));
}

int main() {
    double e, f, g, h, a, b, c;

    printf("Введите e: ");
    scanf("%lf", &e);
    printf("Введите f: ");
    scanf("%lf", &f);
    printf("Введите g: ");
    scanf("%lf", &g);
    printf("Введите h: ");
    scanf("%lf", &h);

    a = calculate_a(e, f);
    b = calculate_b(h, g);
    c = calculate_c(g, h, e);

    printf("a = %lf\n", a);
    printf("b = %lf\n", b);
    printf("c = %lf\n", c);

    return 0;
}

